scripts/copy-documentation-for-new-release.sh

```diff
@@ -32,6 +32,11 @@ function copy_source_to(){
     cp -r "$SOURCE_PATH/release/"* "$DIR"
     cp "$SOURCE_PATH/release.md" "$FILE_PATH"
 
+    # Remove .gitignore'd files from copies
+    # Otherwise they will remain on the local filesytem
+    rm -r "$DIR/examples/node_modules"
+    rm "$DIR/examples/package-lock.json"
+
```

Comment:
See discussion in the [issue thread](https://github.com/sinonjs/sinon/issues/2421#issuecomment-1019615680) and the [commit message](https://github.com/sinonjs/sinon/pull/2426/commits/34a83240c47d6838df2faca9399e562bce84ce98) for more context here - in addition to being untidy and leaving untracked files laying around after we switch back to `master`, they caused the script to fail if I tried to start the same release twice, because then the release folder already existed.

I'm not sure if the latter is a concern, perhaps if you exited with an error from the changelog and thus cancelled the release process? In any case, I think it's worth tidying.

We could tidy them when we create them in the `test-runnable-examples` script as [@fatso83 suggested](https://github.com/sinonjs/sinon/issues/2421#issuecomment-1021310231), but I actually think it's useful to leave behind the files in the `release-source` directory, so that users don't have to reinstall node_modules from scratch every time.

The problem comes when the `node_modules` get copied along with everything else to their archive folder in the `_releases`, where they're never actually being run, so I think cleaning it up here (when we're doing the copying) makes the most sense.